chef/automate
# Desktop Compliance Profiles Page Update

### :nut_and_bolt: Description: What code changed, and why?
Update the Compliance Profile page to include the case of the Desktop Offering is installed. This also updates wording and style for other cases. We want to hide and show the search bar, "X Profiles" "X Available" tabs, profile table, upload button, and text depending on if the Desktop Offering is installed, one or more profiles are installed, and the user has permissions to upload a profile. 

### :chains: Related Resources
#3749

### :+1: Definition of Done
All the below cases look correct. 

### :athletic_shoe: How to Build and Test the Change
1. `build components/automate-ui-devproxy && start_all_services && start_automate_ui_background && ui_logs`
1. Open https://a2-dev.test/compliance/compliance-profiles and ensure it looks like the below first case labeled "Non-Desktop, zero profiles, can upload profiles."
1. Select the "X Available" tab and ![image](https://user-images.githubusercontent.com/1679247/83307733-339ae680-a1ba-11ea-8e9b-b764e7ded02d.png) and profile.
1. Ensure the page looks like the below second case labeled "Non-Desktop, one or more profiles, can upload profiles."
1. Go to https://a2-dev.test/settings/users and create a user named "comp"
1. Go to https://a2-dev.test/settings/policies/compliance-viewer-access#members and add the user comp to the "Compliance Viewers" policy.
1. Log in as the comp user and go to the https://a2-dev.test/compliance/compliance-profiles page
1. Ensure the page looks like the below third case labeled "Non-Desktop, zero profiles, can not upload profiles."
1. Log back in as admin and remove the comp user from the "compliance-viewer-access" policy and add them to the "compliance-editor-access" policy. 
1. Log back in as the comp user and install a profile. 
1. Log back in as the admin user and move the comp user back to only the "compliance-viewer-access" policy. 
1. Log back in as the comp user and go to the profiles page https://a2-dev.test/compliance/compliance-profiles. 
1. Ensure the page looks like the below forth case labeled "Non-Desktop, one or more profiles, can not upload profiles."
1. Now we need to test these four cases with the Desktop offering installed. 
1. Run `enable_desktop` in hab studio and run through the above four cases. 


### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
Cases:
- [x] Non-Desktop, zero profiles, can upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Hide the profiles list
 - Show the upload profiles button.
 - Display the string "Get a profile from the Available tab or upload your own profile to get started!"
![image](https://user-images.githubusercontent.com/1679247/83305410-8c1bb500-a1b5-11ea-993f-d6890867157f.png)

- [x] Non-Desktop, one or more profiles, can upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Show profiles list
 - Show the upload profiles button. 
![image](https://user-images.githubusercontent.com/1679247/83305307-5e367080-a1b5-11ea-9e2b-3f8a47b0ba41.png)

- [x] Non-Desktop, zero profiles, can not upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Hide the profiles list
 - Hide the upload profiles button.
 - Display the string "It looks like no one has uploaded any profiles yet and you don't have permission to upload them. If this is a mistake, then reach out to your administrator."
![image](https://user-images.githubusercontent.com/1679247/83306983-7b207300-a1b8-11ea-903f-dacaac220656.png)

- [x] Non-Desktop, one or more profiles, can not upload profiles.
 - Show search bar
 - Show tabs with the available profiles tab
 - Show profiles list
 - Hide the upload profiles button. 
![image](https://user-images.githubusercontent.com/1679247/83305500-bff6da80-a1b5-11ea-9360-94b6fe8a386c.png)

- [x] Desktop, zero profiles, can upload profiles.
 - Hide tabs with the available profiles tab. 
 - Hide profiles list
 - Show the upload profiles button. 
 - Display the string "Upload a profile to get started!"
![image](https://user-images.githubusercontent.com/1679247/83304584-f5022d80-a1b3-11ea-866a-6ec8af62ee2f.png)

- [x] Desktop, zero profiles, can not upload profile.
 - Hide tabs with the available profiles tab. 
 - Hide profiles list
 - Hide the upload profiles button. 
 - Display the string "It looks like no one has uploaded any profiles yet and you don't have permission to upload them.If this is a mistake, then reach out to your administrator."
![image](https://user-images.githubusercontent.com/1679247/83304753-41e60400-a1b4-11ea-8bf1-6e77002f00c0.png)

- [x] Desktop, one or more profiles, can upload profiles.  
 - Hide tabs with the available profiles tab. 
 - Show profiles list
 - Show the upload profiles button. 
 - Show search bar
![image](https://user-images.githubusercontent.com/1679247/83304625-09462a80-a1b4-11ea-967a-c9c41ceb0ebc.png)

- [x] Desktop, one or more profiles, can not upload profile. 
 - Hide tabs with the available profiles tab. 
 - Show profiles list
 - Hide the upload profiles button. 
 - Show search bar
![image](https://user-images.githubusercontent.com/1679247/83305113-03047e00-a1b5-11ea-8d4e-a6595a51bb76.png)
